extensions/reactive-messaging-http/runtime/src/main/java/io/quarkus/reactivemessaging/http/runtime/ReactiveHttpHandlerBean.java

```diff
@@ -0,0 +1,100 @@
+package io.quarkus.reactivemessaging.http.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.quarkus.reactivemessaging.http.runtime.config.HttpStreamConfig;
+import io.quarkus.reactivemessaging.http.runtime.config.ReactiveHttpConfig;
+import io.quarkus.reactivemessaging.http.runtime.config.WebsocketStreamConfig;
+import io.reactivex.processors.BehaviorProcessor;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ *         <br>
+ *         Date: 01/10/2019
+ */
+@Singleton
+public class ReactiveHttpHandlerBean {
+
+    @Inject
+    ReactiveHttpConfig config;
+
+    private final Map<String, BehaviorProcessor> processors = new HashMap<>();
+    private final Map<String, BehaviorProcessor> websocketProcessors = new HashMap<>();
+
+    public void handleHttp(RoutingContext event) {
+        String path = event.normalisedPath();
+        HttpMethod method = event.request().method();
+        BehaviorProcessor processor = processors.get(key(path, method));
+        if (processor != null) {
+            processor.onNext(new HttpMessage(event.getBody()));
```

Comment:
Here I would create a Message with more metadata, maybe even passing the RoutingContext (that could be retrieved using an `unwrap` method)